src/grader/mutation_test.mli

```diff
@@ -0,0 +1,110 @@
+(* This file is part of Learn-OCaml.
+ *
+ * Copyright (C) 2019 OCaml Software Foundation.
+ * Copyright (C) 2016-2018 OCamlPro.
+ *
+ * Learn-OCaml is distributed under the terms of the MIT license. See the
+ * included LICENSE file for details. *)
+
+type 'a test_result =
+  | Pass
+  | Fail of 'a
+  | Err of exn
+
+(** A mutant is a pair of a string describing the mutant and the
+    mutant function itself.
```

Comment:
Please start with a definition of what a mutant function is, and the problem it solves.